code/modules/admin/view_variables/reference_tracking.dm

```diff
@@ -130,20 +22,49 @@ GLOBAL_LIST_EMPTY(deletion_failures)
 	if(usr?.client)
 		usr.client.running_find_references = type
 
-	testing("Beginning search for references to a [type].")
-	last_find_references = world.time
+	log_reftracker("Beginning search for references to a [type].")
 
-	DoSearchVar(GLOB) //globals
-	for(var/datum/thing in world) //atoms (don't beleive its lies)
-		DoSearchVar(thing, "World -> [thing]")
+	var/starting_time = world.time
+	var/found_ref = FALSE
+
+	//Time to search the whole game for our ref
+	found_ref = DoSearchVar(GLOB, "GLOB", search_time = starting_time) || found_ref //globals
+	log_reftracker("Finished searching globals")
+
+	//Yes we do actually need to do this. The searcher refuses to read weird lists
+	//And global.vars is a really weird list
+	var/global_vars = list()
+	for(var/key in global.vars)
+		global_vars[key] = global.vars[key]
+
+	found_ref = DoSearchVar(global_vars, "Native Global", search_time = starting_time) || found_ref
+	log_reftracker("Finished searching native globals")
 
 	for(var/datum/thing) //datums
-		DoSearchVar(thing, "World -> [thing]")
+		found_ref = DoSearchVar(thing, "Datums -> [thing.type]", search_time = starting_time) || found_ref
+	log_reftracker("Finished searching datums")
 
+	//Warning, attempting to search clients like this will cause crashes if done on live. Watch yourself
+#ifndef REFERENCE_DOING_IT_LIVE
 	for(var/client/thing) //clients
-		DoSearchVar(thing, "World -> [thing]")
+		found_ref = DoSearchVar(thing, "Clients -> [thing.type]", search_time = starting_time) || found_ref
+	log_reftracker("Finished searching clients")
+#endif
+
+#ifdef REFERENCE_TRACKING_FAST
+	if(found_ref)
+		log_reftracker("Skipped searching atoms (other ref(s) found)")
+	else
+		for(var/datum/thing in world) //atoms (don't beleive its lies)
+			found_ref = DoSearchVar(thing, "World -> [thing.type]", search_time = starting_time) || found_ref
+		log_reftracker("Finished searching atoms")
```

Comment:
```suggestion
		for(var/datum/thing in world) //atoms (don't beleive its lies)
			found_ref = DoSearchVar(thing, "World -> [thing.type]", search_time = starting_time)
			if(found_ref)
				log_reftracker("Aborting atoms search (ref found)")
				break
		log_reftracker("Finished searching atoms")
```

Works locally and it made every atom search I saw take less than a second.

Reply:
I really wish lummox would give us a way of detecting how many refs there are remaining

Reply:
@LemonInTheDark Honestly I'd rather fix them one at a time than sit through the entirety of the atom search for each harddel while bee still has several.

But yes it would be nice if he exposed the refcount.